src/main/java/eu/europeana/apikey/controller/ApikeyController.java

```diff
@@ -0,0 +1,559 @@
+     * HTTP 410 when the requested Apikey is deprecated (i.e. has a past deprecationdate)
+     * <p>
+     * Addionally, the field 'Apikey-not-found' containing the string "apikey-not-found" will be available in the
+     * response header to help telling this HTTP 404 apart from one returned by the webserver for other reasons
+     */
+    @CrossOrigin(maxAge = 600)
+    @PutMapping(path = "/{id}/disable")
+    public ResponseEntity disable(@PathVariable("id") String id) throws ApiKeyException {
+        LOG.debug("Disabling API key {}...", id);
+
+        KeycloakAuthenticationToken kcAuthToken = checkManagerCredentials();
+        Apikey                      apikey      = checkKeyExists(id);
+        checkKeyDeprecated(apikey);
+
+        if (isNoKeycloakRequest(kcAuthToken)) {
+            keycloakManager.disableClient(id, (KeycloakSecurityContext) kcAuthToken.getCredentials());
+        }
+        apikey.setDeprecationDate(new DateTime(DateTimeZone.UTC).toDate());
+        this.apikeyRepo.save(apikey);
+        return new ResponseEntity(HttpStatus.NO_CONTENT);
+    }
+
+    /**
+     * Re-enables a given invalid Apikey (of which the deprecationdate column has been set to a past time).
+     * NOTE that for now the code does not check if the key is really deprecated; it merely executes the update (if any)
+     * in such cases.
+     *
+     * @param id the apikey to re-enable
+     * @return JSON response containing the fields annotated with @JsonView(View.Public.class) in apikey.java
+     * HTTP 200 upon successful Apikey update
+     * HTTP 400 when a required parameter is missing or has an invalid value
+     * HTTP 401 in case of an invalid request
+     * HTTP 403 if the request is unauthorised
+     * HTTP 404 if the apikey is not found
+     * HTTP 406 if a response MIME type other than application/JSON was requested
+     * HTTP 415 if the submitted request does not contain a valid JSON body
+     */
+    @CrossOrigin(maxAge = 600)
+    @PutMapping(path = "/{id}/enable")
+    public Apikey enable(@PathVariable("id") String id) throws ApiKeyException {
+        LOG.debug("Enabling API key {}...", id);
+
+        KeycloakAuthenticationToken kcAuthToken = checkManagerCredentials();
+        Apikey                      key         = checkKeyExists(id);
+        if (key.getDeprecationDate() == null) {
+            throw new ApiKeyNotDeprecatedException(id);
+        }
+
+        if (isNoKeycloakRequest(kcAuthToken)) {
+            // call Keycloak update only when this request does not come from Keycloak
+            keycloakManager.enableClient(id, (KeycloakSecurityContext) kcAuthToken.getCredentials());
+        }
+
+        // remove deprecationdate: this enables the key again
+        key.setDeprecationDate(null);
+        this.apikeyRepo.save(key);
+        return key;
+    }
+
+    private boolean isNoKeycloakRequest(KeycloakAuthenticationToken keycloakAuthenticationToken) {
```

Comment:
I think it's better to not use negations in methods names because this quickly becomes confusing (and doesn't adhere to bean standards). I would prefer to continue using isKeycloakRequest and simply negate the result.